src/app/projects/projlist-map/projlist-map.component.ts

```diff
@@ -270,12 +270,14 @@ export class ProjlistMapComponent implements AfterViewInit, OnChanges, OnDestroy
     deletedApps.forEach(app => {
       const markerIndex = _.findIndex(this.markerList, { projectId: app._id });
       const shapefileIndex = _.findIndex(this.shapefileList, { projectId: app._id });
+
       if (markerIndex >= 0) {
         const markers = this.markerList.splice(markerIndex, 1);
         this.markerClusterGroup.removeLayer(markers[0]);
       }
       if (shapefileIndex >= 0) {
-        this.shapefileList.splice(shapefileIndex, 1);
+        const shapefiles = this.shapefileList.splice(shapefileIndex, 1);
+        this.map.removeLayer(shapefiles[0]);
```

Comment:
There's the magic line that changes everything! Nice work figuring this one out.